ledger/src/blockstore_meta.rs

```diff
@@ -14,8 +14,18 @@ use {
 
 bitflags! {
     #[derive(Default, Serialize, Deserialize)]
+    /// Flags to indicate whether a slot is a descendant of a sot on the main fork
     pub struct ConnectedFlags:u8 {
+        // A slot can be connected in two scenarios:
+        // 1) If the parent is connected and if the slot is full. By this
```

Comment:
>         // A slot can be connected in two scenarios:
>         // 1) If the parent is connected and if the slot is full. By this

I think you're probably defining something based on its own definition xD.  How about something like the following:

```
A slot is connected if it is:
* A root, or
* is *full* and its parent is connected.
```

In other words, a connected slot is a slot where all slots within [most recent root, slot] are full.
And we probably want to briefly define what is full in the comment as well.

If it is somehow possible to have a root not connected to either snapshot or genesis, then we can replace root by snapshot or genesis in the above statement.

Reply:
Pushed some updates, let me know what you think